dateparser/date.py

```diff
@@ -332,7 +338,7 @@ class DateDataParser:
 
     @apply_settings
     def __init__(self, languages=None, locales=None, region=None, try_previous_locales=False,
-                 use_given_order=False, settings=None):
+                 use_given_order=False, settings=None, detect_languages_function=None):
 
         if languages is not None and not isinstance(languages, (list, tuple, Set)):
```

Comment:
Based on the usage of `self.languages` further I think this should only accept list.
Ref https://github.com/scrapinghub/dateparser/pull/932/files#r679906290

Reply:
Yes, that will be great but maybe that is a breaking change (I have not checked) more so the type check raises TypeError with the message `languages argument must be a list`. 

Please suggest and I think this should be a separate micro PR.

@lopuhin @noviluni

Reply:
As you mentioned, changing the input would be a breaking change. Currently, it works with lists, sets and tuples:

```python
>>> dateparser.parse("12/12/2000", languages=["en"])
datetime.datetime(2000, 12, 12, 0, 0)

>>> dateparser.parse("12/12/2000", languages={"en"})
datetime.datetime(2000, 12, 12, 0, 0)

>>> dateparser.parse("12/12/2000", languages=("en",))
datetime.datetime(2000, 12, 12, 0, 0)
```

However, you can fix this without changing the input by casting it to a list. So if you change this line like this:

```python
self.languages = list(languages) if languages else None
```

it should work. Let me know if it's not :)

Reply:
thanks, I have made the changes

Reply:
Correct me if I am wrong but I think this change is still pending.

Reply:
Hi @kishan3, these changes were part of this commit [here](https://github.com/scrapinghub/dateparser/pull/932/commits/6e6eb14cfb289cce73a1e4276d02d99e51ed253f) 

In dateparser/date.py, line 368

Reply:
:+1: